core/src/test/scala/caliban/execution/ExecutionSpec.scala

```diff
@@ -348,6 +349,19 @@ object ExecutionSpec extends DefaultRunnableSpec {
 
         assertM(interpreter.flatMap(_.execute(query)).map(_.data.toString))(equalTo("""{"test":{"a":333}}"""))
       },
+      testM("Play Json scalar") {
+        import play.api.libs.json._
+        import caliban.interop.play.json._
+        case class Queries(test: JsValue)
+
+        val interpreter = graphQL(RootResolver(Queries(Json.obj(("a", JsNumber(333)))))).interpreter
+        val query       = gqldoc("""
+             {
+               test
+             }""")
+
+        assertM(interpreter.flatMap(_.execute(query)).map(_.data.toString))(equalTo("""{"test":{"a":333}}"""))
```

Comment:
Could you also add a field that returns `None` so that there is a regression for `nullable` fields.

Reply:
There's already one in `GraphQLResponseSpec` and it actually fails when I just put option values instead of separating optional fields.
I checked about None-fields behavior in play and it seems that it doesn't remove null fields by default. Moreover, it seems there's not simple way to turn that off - you have to manually process json AST before printing to do it.

Reply:
To expand - the behavior you describe works for macro-derived codecs. If you create object manually and specify a field to be None you still get a null.
See an example:

https://scastie.scala-lang.org/oCuWojs4TjajMl0laRYKSg

Reply:
I just used an intermediate dto class with derived codec so that these rules apply. Now it looks even better.